pkg/controllers/user/multiclusterapp/controller.go

```diff
@@ -0,0 +1,114 @@
+	AppClient             pv3.AppInterface
+	//AppRevisionGetter     v3.AppRevisionsGetter
+	//AppGetter             v3.AppsGetter
+	NsClient corev1.NamespaceInterface
+	NsLister corev1.NamespaceLister
+}
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	m := &MultiClusterAppController{
+		//AppGetter:             user.Management.Project,
+		//AppRevisionGetter:     user.Management.Project,
+		AppClient:             cluster.Management.Project.Apps(""),
+		NsClient:              cluster.Management.Core.Namespaces(""),
+		NsLister:              cluster.Management.Core.Namespaces("").Controller().Lister(),
+		MultiClusterAppClient: cluster.Management.Management.MultiClusterApps(""),
+		MultiClusterAppLister: cluster.Management.Management.MultiClusterApps("").Controller().Lister(),
+	}
+	m.MultiClusterAppClient.AddHandler(ctx, "multi-cluster-app-controller", m.sync)
+}
+
+func (m *MultiClusterAppController) sync(key string, mcapp *v3.MultiClusterApp) (runtime.Object, error) {
+	if mcapp == nil || mcapp.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	var answerMap map[string]map[string]string
+	var answerFound bool
+
+	// create app object spec
+	// create the externalId field
+	externalId := fmt.Sprintf("catalog://?catalog=%s&template=%s&version=%s", mcapp.Spec.CatalogName, mcapp.Spec.TemplateName, mcapp.Spec.TemplateVersionName)
+
+	// create a map, where key is the projectID or clusterID, or "global" if neither is provided, and value is the actual answer values
+	answerMap = make(map[string]map[string]string)
+	for _, a := range mcapp.Spec.Answers {
+		if a.ProjectName != "" {
+			answerMap[a.ProjectName] = make(map[string]string)
+			answerMap[a.ProjectName] = a.Values
+		} else if a.ClusterName != "" {
+			answerMap[a.ClusterName] = make(map[string]string)
+			answerMap[a.ClusterName] = a.Values
+		} else {
+			answerMap[globalScopeAnswersKey] = make(map[string]string)
+			answerMap[globalScopeAnswersKey] = a.Values
+		}
+	}
+
+	// for all targets, create the App{} instance, so that helm controller App lifecycle can pick it up
+	for _, t := range mcapp.Spec.Targets {
+		app := pv3.App{
+			Spec: pv3.AppSpec{
+				ProjectName:         t.Spec.ProjectName,
+				TargetNamespace:     t.Spec.TargetNamespace,
+				ExternalID:          externalId,
+				MultiClusterAppName: mcapp.Name,
+			},
+		}
+
+		if ans, ok := answerMap[t.Spec.ProjectName]; ok {
+			app.Spec.Answers = ans
+			answerFound = true
+		}
+
+		if !answerFound {
+			// find the answers for the cluster of this project
+			split := strings.SplitN(t.Spec.ProjectName, ":", 2)
+			if len(split) != 2 {
+				return nil, fmt.Errorf("error in splitting project name")
+			}
+			clusterName := split[0]
+			if ans, ok := answerMap[clusterName]; ok {
+				app.Spec.Answers = ans
+				answerFound = true
+			}
+
+			if !answerFound {
+				if ans, ok := answerMap[globalScopeAnswersKey]; ok {
+					app.Spec.Answers = ans
+					answerFound = true
```

Comment:
where is answerFound parameter being used?

Reply:
Yeah guess there's no need to set it to true at this point